test/test_nn.py

```diff
@@ -805,6 +805,18 @@ def __init__(self):
         self.assertEqual(names(m.named_buffers(remove_duplicate=False)),
                          ["buffer1", "buffer2"])
 
+        class M(nn.Module):
```

Comment:
Could you also have a test for weight being shared because the same Module is re-used
```
self.mod1 = nn.Linear()
self.mod2 = self.mod1
```

Reply:
Added